lib/freeclimb/utils/request_verifier.rb

```diff
@@ -0,0 +1,58 @@
+module Freeclimb
+    class RequestVerifier
+        @@DEFAULT_TOLERANCE = 5 * 60 * 1000
+        attr_accessor :info
+
+        def verifyRequestSignature(requestBody, requestHeader, signingSecret, tolerance=DEFAULT_TOLERANCE)
+            checkRequestBody(requestBody)
+            checkRequestHeader(requestHeader)
+            checkSigningSecret(signingSecret)
+            checkTolerance(tolerance)
+            @info = Freeclimb::SignatureInformation.new(requestHeader)
+            verifyTolerance(@info, tolerance)
+            verifySignature(@info, requestBody, signingSecret)
+        end
+        
+        def checkRequestBody(requestBody)
+            if requestBody == "" or requestBody == nil
+                raise 'Request Body cannot be empty or null'
+            end
+        end
+
+        def checkRequestHeader(requestHeader)
+            if requestHeader == "" or requestHeader == nil
+                raise 'Error with request header, Request header is empty'
+            elsif ! requestHeader.include? "t"
+                raise 'Error with request header, timestamp is not present'
+            elsif ! requestHeader.include? "v1"
+                raise 'Error with request header, signatures are not present'
+            end
+        end
+
+        def checkSigningSecret(signingSecret)
+            if signingSecret == "" or signingSecret == nil
+                raise 'Signing secret cannot be empty or null'
+            end
+        end
+
+        def checkTolerance(tolerance)
+            fixnum = (2**(0.size * 8 -2) -1)
+            if tolerance <= 0 or tolerance == Float::NAN or tolerance >= fixnum
```

Comment:
This is incorrect use of ruby `or`. http://phrogz.net/programmingruby/language.html#table_18.4 
The `or` `and` `not` keywords are bad smells with ruby. 
```
if tolerance <= 0 || !(tolerance.is_a? Integer)
```
is a more appropriate conditional

Reply:
will change in latest commit